### ISSUE TITLE
[Feature] Contribution graph

### ISSUE THREAD

**Issue by sthewissen** [sthewissen/MVP#21]
A GitHub type contribution graph would be really cool to have :)

![image](https://user-images.githubusercontent.com/2419439/84744238-40d61480-afb3-11ea-907f-9eaca65847e9.png)

**Comment by sthewissen** [comment]
@krdmllr I agree it would be really cool. The biggest problem is the data. The MVP API is a notoriously outdated thing that doesn't really follow any modern API conventions. Getting the actual data to compile these statistics is a tough chore to do. I think the best bet is to:

- Make a separate statistics tab.
- Initially gather all the data, which will involve multiple API calls depending on how many contributions the person has made.
- Cache the statistics data locally so the next visit to the tab will be smooth(er).

The problem is, when do we invalidate this cache? Or do we only allow the user to force update the statistics manually? Theoretically, someone can add MVP activities from a year in the past, so these items would not show up if we just get the first page of new data for example. We'd always have to get all the data to see if something changed.

I chose a day view because it's not necessarily commits we're talking about here. It's any activity for the MVP program, so it can be a blog post, a podcast, a session at a conference, etc. The idea was to basically have the day view and only have a yes/no color if you did something that day. That way you can see what months you were more active.

**Comment by krdmllr** [comment]
A combination of force reload + adding data whenever the user adds data trough the app + update data the user loads in the activity feed could work then? Do you know if the api allows that much load and how fast it is?

**Comment by krdmllr** [comment]
Should we give it a try or add a "wait-for-new-api" tag?

**Comment by sthewissen** [comment]
There is practically no one at Microsoft who seems to be responsible for the API. I've tried getting improvements in there, but it took over half a year if not more the last time I did.
